dsgrid/registry/project_registry_manager.py

```diff
@@ -239,36 +279,69 @@ def _submit_dataset(
             project_config.config_id,
         )
 
-    # def check_dataset_base_to_project_base_mappings(self, project_config, dataset_config, dataset):
-    #    """Check that the dataset contains all mappings of base dimensions.
-
-    #    Parameters
-    #    ----------
-    #    project_config : ProjectConfig
-    #    dataset_config : DatasetConfig
-    #    dataset : Dataset
-
-    #    Raises
-    #    ------
-    #    DSGInvalidDimensionMapping
-    #        Raised if a requirement is violated.
-
-    #    """
-    #    needs_full_join = set()
-    #    types = [x for x in DimensionType if x != DimensionType.TIME]
-    #    for type1, type2 in itertools.product(types, types):
-    #        if type1 != type2:
-    #            needs_full_join.add((type1, type2))
-
-    #    for mapping_ref in project_config.model.dimension_mappings.base_to_base:
-    #        from_dimension = mapping_ref.from_dimension_type
-    #        to_dimension = mapping_ref.to_dimension_type
-    #        needs_full_join.remove((from_dimension, to_dimension))
-    #        mapping_config = self._dimension_mapping_mgr.get_by_id(mapping_ref.mapping_id)
-    #        mapping_records = models_to_dataframe(mapping_config.model.records)
-    #        # TODO perform checks across mapping_records, dataset.load_data, dataset.load_data_lookup
-
-    #    # TODO Perform checks on all from-to dimensions in needs_full_join.
+    @track_timing(timer_stats_collector)
+    def _check_dataset_base_to_project_base_mappings(
+        self,
+        project_config: ProjectConfig,
+        dataset_config: DatasetConfig,
+        mapping_references: List[DimensionMappingReferenceModel],
+    ):
+        """Check that a dataset has all project-required dimension records."""
+        handler = make_dataset_schema_handler(
+            dataset_config, self._dimension_mgr, self._dimension_mapping_mgr
+        )
+        pivot_dimension = handler.get_pivot_dimension_type()
+        exclude_dims = set([DimensionType.TIME, DimensionType.DATA_SOURCE, pivot_dimension])
+        types = [x for x in DimensionType if x not in exclude_dims]
+        dimension_pairs = [tuple(sorted((x, y))) for x, y in itertools.combinations(types, 2)]
+
+        data_source = dataset_config.model.data_source
+        associations = project_config.dimension_associations
+        dim_table = handler.get_unique_dimension_rows(mapping_references)
+        for type1, type2 in dimension_pairs:
+            records = associations.get_associations_by_data_source(data_source, type1, type2)
```

Comment:
Will this grab the right associations if we have something like:

| data_source | sector |
|------------- | ------ |
| resstock | res |

| sector | subsector |
| ------ | ---------- |
| res      | SingleFamily |
| res     | MultiFamiliy |

?

Reply:
Does this answer your question? Also, you can directly load the dimension associations for StandardScenarios without project or datasets if you'd like to experiment.
```
(Pdb) self.get_associations(DimensionType.DATA_SOURCE, DimensionType.SECTOR).show()
+-----------+------+
|data_source|sector|
+-----------+------+
|   comstock|   com|
|   comstock|   res|
+-----------+------+
(Pdb) self.get_associations(DimensionType.SECTOR, DimensionType.SUBSECTOR).show()
+------+--------------------+
|sector|           subsector|
+------+--------------------+
|   com|FullServiceRestau...|
|   com|            Hospital|
|   com|          LargeHotel|
|   com|         LargeOffice|
|   com|        MediumOffice|
|   com|          Outpatient|
|   com|       PrimarySchool|
|   com|QuickServiceResta...|
|   com|          SmallHotel|
|   com|         SmallOffice|
|   com|    StandaloneRetail|
|   com|           StripMall|
|   com|           Warehouse|
|   res|    MidriseApartment|
+------+--------------------+
(Pdb) self.get_associations_by_data_source("comstock", DimensionType.SECTOR, DimensionType.SUBSECTOR).show()
+--------------------+------+
|           subsector|sector|
+--------------------+------+
|FullServiceRestau...|   com|
|            Hospital|   com|
|          LargeHotel|   com|
|         LargeOffice|   com|
|        MediumOffice|   com|
|          Outpatient|   com|
|       PrimarySchool|   com|
|QuickServiceResta...|   com|
|          SmallHotel|   com|
|         SmallOffice|   com|
|    StandaloneRetail|   com|
|           StripMall|   com|
|           Warehouse|   com|
|    MidriseApartment|   res|
+--------------------+------+
```